File: Data Structures & Algorithms/singlyLinkedList/submission-1.cs
public class LinkedList {

    private Node head;
    private Node tail;
    private int count;

    public LinkedList() {
        this.head = null;
        this.tail = null;
        this.count = 0;
    }

    public int Get(int index) {
        if(index >= count) return -1;
        Node curr = new Node(0);
        curr = this.head;
        for(int i = 0; i < index; ++i) {
            curr = curr.next;
        }
        return curr.val;
    }

    public void InsertHead(int val) {
        Node newNode = new Node(val);
        newNode.next = this.head;
        this.head = newNode;
        if(head.next == null) this.tail = newNode;
        count++;
    }

    public void InsertTail(int val) {
        Node newNode = new Node(val);
        if (tail == null) {
            this.tail = newNode;
            this.head = newNode;
        }

        else {
            this.tail.next = newNode;
            this.tail = newNode;
        }
        count++;
    }

    public bool Remove(int index) {
        if(index >= count) return false;
        if(index == 0) {
            head = head.next;
            count--;
            return true;
        }

        Node curr = new Node(0);
        curr = this.head;
        for(int i = 1; i < index; ++i) {
            curr = curr.next;
        }
        if(curr.next == tail) tail = curr;
        curr.next = curr.next.next;
        count--;
        return true;
    }

    public List<int> GetValues() {
        Node curr = new Node(0);
        curr = this.head;
        List<int> values = new List<int>();
        for(int i = 0; i < count; ++i) {
            values.Add(curr.val);
            curr = curr.next;
        }
        return values;
    }
}

public class Node {
    public int val;
    public Node next;

    public Node(int val) {
        this.val = val;
        this.next = null;
    }
}

File: Data Structures & Algorithms/singlyLinkedList/submission-1.py
'''
    Comment: Took Help from Claude on How to improve
'''

from typing import List, Optional

class ListNode:
    def __init__(self, val: int = 0, next: "ListNode | None" = None):
        self.val = val
        self.next: Optional[ListNode] = next


class LinkedList:
    def __init__(self):
        self.head: Optional[ListNode] = ListNode()
        self.tail: Optional[ListNode] = self.head

    def get(self, index: int) -> int:
        curr: Optional[ListNode] = self.head.next
        i: int = 0

        while curr:
            if i == index:
                return curr.val
            curr = curr.next
            i += 1
        return -1

    def insertHead(self, val: int) -> None:
        node: ListNode = ListNode(val, self.head.next)
        self.head.next = node
        if node.next is None:
            self.tail = node

    def insertTail(self, val: int) -> None:
        node: ListNode = ListNode(val)
        self.tail.next = node
        self.tail = node

    def remove(self, index: int) -> bool:
        prev: ListNode = self.head
        i = 0
        while prev.next and i!=index:
            prev = prev.next
            i += 1
        if prev.next is None:
            return False
        if prev.next is self.tail:
            self.tail = prev
        prev.next = prev.next.next
        return True

    def getValues(self) -> List[int]:
        curr: Optional[ListNode] = self.head.next
        nodes: List[int] = []

        while curr:
            nodes.append(curr.val)
            curr = curr.next

        return nodes
